src/plugins/intel_cpu/src/emitters/snippets/aarch64/kernel_executors/gemm_copy_b.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "gemm_copy_b.hpp"
+
+#include "openvino/core/parallel.hpp"
+#include "snippets/utils/utils.hpp"
+#include "transformations/snippets/aarch64/op/gemm_utils.hpp"
+#include "transformations/tpp/common/op/brgemm.hpp"
+
+namespace ov::intel_cpu::aarch64 {
+
+GemmCopyBKernelKaiConfig::GemmCopyBKernelKaiConfig(const size_t n_blk_size) : m_n_blk_size(n_blk_size) {}
+
+GemmCopyBKernelKaiConfig& GemmCopyBKernelKaiConfig::operator=(GemmCopyBKernelKaiConfig other) {
+    m_N = other.get_N();
+    m_K = other.get_K();
+    m_hash = compute_hash();
+    return *this;
+}
+
+bool GemmCopyBKernelKaiConfig::is_completed() const {
+    return !ov::snippets::utils::one_of(0ul, m_N, m_K, m_n_blk_size);
+}
+#ifdef SNIPPETS_DEBUG_CAPS
+#    define PRINT(X) ss << #X << " = " << (X) << "\n"
+std::string GemmCopyBKernelKaiConfig::to_string() const {
+    std::stringstream ss;
+    PRINT(m_N);
+    PRINT(m_K);
+    PRINT(m_n_blk_size);
+    return ss.str();
+}
+#    undef PRINT
+#endif
+
+void GemmCopyBKernelKaiConfig::update(size_t N, size_t K) {
+    // If one of the dims is zero, it means that GemmCopyB won't be executed (in Loop with work_amount = 0, for
+    // example) To process this case, we have to make this Config as empty (nullify runtime parameters)
+    if (ov::snippets::utils::one_of(0ul, N, K)) {
+        m_N = 0;
+        m_K = 0;
+    } else {
+        m_N = N;
+        m_K = K;
+    }
+    m_hash = compute_hash();
+}
+
+size_t GemmCopyBKernelKaiConfig::compute_hash() const {
+    size_t seed = 0;
+    seed = dnnl::impl::hash_combine(seed, m_N);
+    seed = dnnl::impl::hash_combine(seed, m_K);
+    seed = dnnl::impl::hash_combine(seed, m_n_blk_size);
+    return seed;
+}
+
+GemmCopyBKaiKernelExecutor::GemmCopyBKaiKernelExecutor(GemmCopyBKernelKaiConfig config)
+    : snippets::KernelExecutor<GemmCopyBKernelKaiConfig, GemmCopyBCompiledKernel>(std::move(config)) {}
+
+void GemmCopyBKaiKernelExecutor::update_kernel(const GemmCopyBKernelKaiConfig& config,
+                                               std::shared_ptr<GemmCopyBCompiledKernel>& kernel) const {
+    if (kernel == nullptr) {
```

Comment:
Minor: could you please add a comment on why we create kernel only once? If I am not mistaken, this is the first shape agnostic kernel in snippets, so this code may raise some questions.